src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AutomaticCaseCreationTest.java

```diff
@@ -0,0 +1,166 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.dm.DocumentManagementUploadService;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CaseSearcher;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.EnvelopeMessager;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CaseDataExtractor.getCaseDataForField;
+
+@SpringBootTest
+@ActiveProfiles("nosb")  // no servicebus queue handler registration
+@Disabled
+public class AutomaticCaseCreationTest {
+
+    private static final String SERVICE_CASE_JURISDICTION = "BULKSCAN";
+    private static final String SERVICE_CASE_TYPE_ID = "Bulk_Scanned";
+
+    // case type ID for exception records related to service which is disabled for automatic case creation
+    private static final String DISABLED_SERVICE_EXCEPTION_RECORD_CASE_TYPE_ID = "BULKSCAN_ExceptionRecord";
+
+    // case type ID for exception records related to service which is enabled for automatic case creation
+    private static final String ENABLED_SERVICE_EXCEPTION_RECORD_CASE_TYPE_ID = "BULKSCANAUTO_ExceptionRecord";
+
+
+    @Autowired
+    private CaseSearcher caseSearcher;
+
+    @Autowired
+    private EnvelopeMessager envelopeMessager;
+
+    @Autowired
+    private DocumentManagementUploadService dmUploadService;
+
+    private String documentUrl;
+
+    @BeforeEach
+    void setup() {
+        documentUrl = dmUploadService.uploadToDmStore(
+            "Certificate.pdf",
+            "documents/supplementary-evidence.pdf"
+        );
+    }
+
+    @Test
+    void should_create_case_when_envelope_data_is_valid_and_service_is_enabled_for_auto_creation() throws Exception {
+        // when
+        var poBox = UUID.randomUUID();
+        String envelopeId = sendEnvelopeMessage("envelopes/valid-new-application-bulkscanauto.json", poBox);
+
+        // then
+        await("Wait for service case to be created")
```

Comment:
can we add `envelopeId` for easier debug?

Reply:
good idea, will do

Reply:
done